lib/ansible/modules/utilities/logic/async_wrapper.py

```diff
@@ -232,14 +233,13 @@ def _run_module(wrapped_cmd, jid, job_path):
     jobdir = os.path.expanduser(async_dir)
     job_path = os.path.join(jobdir, jid)
 
-    if not os.path.exists(jobdir):
-        try:
-            os.makedirs(jobdir)
-        except Exception:
-            print(json.dumps({
-                "failed": 1,
-                "msg": "could not create: %s" % jobdir
-            }))
+    try:
+        makedirs_safe(jobdir)
+    except Exception as e:
+        print(json.dumps({
+            "failed": 1,
+            "msg": "could not create %s: %s" % (jobdir, to_native(e))
```

Comment:
you want to_text here as it is a json string, use to_native for when raising an exception